pkg/kubelet/api/v1alpha1/runtime/api.proto

```diff
@@ -707,6 +729,8 @@ message Image {
     repeated string repo_digests = 3;
     // The size of the image in bytes.
     optional uint64 size = 4;
+    // The uid that will run the command(s).
+    optional int64 uid = 5;
```

Comment:
@euank @yifan-gu are you ok with this or do you prefer runtime itself checks the `run_as_non_root` option? 

Background: In the initial version, kubelet set the `run_as_non_root` field in the security context and pass it to the runtime (via CRI). Runtime checks the (docker) image and verify that the user is not set to root. This assumes that the `RuntimeService` is aware of the images and can perform checks. Since we may want to decouple `ImageService` and `RuntimeService` in the future, I suggested move up this logic and let kubelet perform the check. This requires reporting the user name or ID as part of the image for kubelet to check.